### PR TITLE
add ALI flags

### PR DESCRIPTION
**Pull request by LilyFaFa**
This is part of #4127.

### CODE REVIEW

File: protokube/cmd/protokube/main.go [hunk 1]
```diff
@@ -194,7 +194,14 @@ func run() error {
 		if clusterID == "" {
 			clusterID = osVolumes.ClusterID()
 		}
-
+	} else if cloud == "alicloud" {
+		glog.Info("Initializing AliCloud volumes")
+		aliVolumes, err := protokube.NewALIVolumes()
+		if err != nil {
+			glog.Errorf("Error initializing Aliyun: %q", err)
+			os.Exit(1)
```
Comment (chrisz100) [inline protokube/cmd/protokube/main.go]: You sure exiting entirely is desired?

**Comment by chrisz100** [comment]
As this is a feature and we push to get 1.12 out we'll handle this with 1.13

/milestone 1.13

File: protokube/cmd/protokube/main.go (hunk 1 above)
Comment (justinsb) [reply protokube/cmd/protokube/main.go]: I think it's what the other providers do - failure to initialize the cloud if we've been told to do it is pretty catastrophic.

That said, yes, we're probably overdue for a refactor of this function to return error!

File: upup/pkg/fi/cloudup/tagbuilder.go [hunk 2]
```diff
@@ -60,6 +60,10 @@ func buildCloudupTags(cluster *api.Cluster) (sets.String, error) {
 
 	case api.CloudProviderOpenstack:
 
+	case api.CloudProviderALI:
+		{
+			tags.Insert("_ali")
```
Comment (justinsb) [inline upup/pkg/fi/cloudup/tagbuilder.go]: I hope you don't need this - we're trying to get rid of the tags, and I don't see you using them anywhere :-)

But ... it's pretty harmless!

**Comment by justinsb** [comment]
I'm fine with merging this into 1.12 as it is scoped to ali cloud, which is feature-gated.  Also looks like a few bug fixes in the mix...

/approve
/lgtm

**Pull request merged**